novawallet/Modules/NetworkNode/NetworkNodeViewFactory.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+import SoraFoundation
+
+struct NetworkNodeViewFactory {
+    static func createNodeAddView(with chainId: ChainModel.Id) -> NetworkNodeViewProtocol? {
+        let connectionFactory = ConnectionFactory(
+            logger: Logger.shared,
+            operationQueue: OperationManagerFacade.assetsSyncQueue
+        )
+
+        let chainRegistry = ChainRegistryFactory.createDefaultRegistry()
+
+        let repository = SubstrateRepositoryFactory().createChainRepository()
+        
+        let operationQueue: OperationQueue = {
+            let operationQueue = OperationQueue()
+            operationQueue.qualityOfService = .userInitiated
+            return operationQueue
+        }()
+        
+        let interactor = NetworkNodeAddInteractor(
+            chainRegistry: chainRegistry,
+            connectionFactory: connectionFactory,
+            chainId: chainId,
+            repository: repository,
+            operationQueue: operationQueue
+        )
+        
+        let wireframe = NetworkNodeWireframe()
+        
+        let networkViewModelFactory = NetworkViewModelFactory()
+
+        let presenter = NetworkNodeAddPresenter(
+            interactor: interactor,
+            wireframe: wireframe,
+            networkViewModelFactory: networkViewModelFactory,
+            localizationManager: LocalizationManager.shared
+        )
+
+        let view = NetworkNodeViewController(
+            presenter: presenter,
+            localizationManager: LocalizationManager.shared
+        )
+
+        presenter.view = view
+        interactor.presenter = presenter
+
+        return view
+    }
+    
+    static func createNodeEditView(
+        with chainId: ChainModel.Id,
+        _ node: ChainNodeModel
+    ) -> NetworkNodeViewProtocol? {
+        let connectionFactory = ConnectionFactory(
+            logger: Logger.shared,
+            operationQueue: OperationManagerFacade.assetsSyncQueue
+        )
+
+        let chainRegistry = ChainRegistryFactory.createDefaultRegistry()
```

Comment:
```ChainRegistryFacade.sharedRegistry```. Here we are creating a new instance instead of using the singleton. Please, make sure we are not doing the same in other places.